src/NumericalAlgorithms/Interpolation/InterpolationTargetLineSegment.hpp

```diff
@@ -0,0 +1,127 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <algorithm>
+#include <pup.h>
+
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "DataStructures/DataBox/DataBoxTag.hpp"
+#include "NumericalAlgorithms/Interpolation/SendPointsToInterpolator.hpp"
+#include "NumericalAlgorithms/Interpolation/Tags.hpp"
+#include "Parallel/ConstGlobalCache.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+namespace intrp {
+
+namespace OptionHolders {
+/// A line segment extending from `Begin` to `End`, containing `NumberOfPoints`
+/// uniformly-spaced points including the endpoints.
+template <size_t VolumeDim>
+struct LineSegment {
+  struct Begin {
+    using type = std::array<double, VolumeDim>;
+    static constexpr OptionString help = {"Begin"};
+  };
+  struct End {
+    using type = std::array<double, VolumeDim>;
+    static constexpr OptionString help = {"End"};
+  };
+  struct NumberOfPoints {
+    using type = size_t;
+    static constexpr OptionString help = {"NumberOfPoints"};
+    static type lower_bound() { return 2; }
+  };
+  using options = tmpl::list<Begin, End, NumberOfPoints>;
+  static constexpr OptionString help = {"Options for LineSegment"};
```

Comment:
"A line segment extending from Begin to End, containing NumberOfPoints uniformly-spaced points including the endpoints."